php/classes/handlers/class-castos-handler.php

```diff
@@ -362,15 +372,56 @@ public function upload_podcast_to_podmotor( $post ) {
 	}
 
 	/**
-	 * Gets the featured image url
+	 * Gets podcast image url
 	 *
 	 * @param $post
-	 * @param $post_body
 	 *
-	 * @return mixed
+	 * @return string
+	 */
+	public function get_podcast_image_url( $post ) {
+		$key    = 'cover_image';
+		$id_key = 'cover_image_id';
+
+		$podcast_image = filter_input( INPUT_POST, $key, FILTER_VALIDATE_URL );
+		$attachment_id = filter_input( INPUT_POST, $id_key );
+
+		if ( ! $podcast_image || ! $this->is_valid_podcast_image( $attachment_id ) ) {
+			$podcast_image = get_post_meta( $post->ID, $key );
+			$attachment_id = get_post_meta( $post->ID, $id_key );
+		}
+
+		if ( ! $podcast_image || ! $this->is_valid_podcast_image( $attachment_id ) ) {
+			$podcast_image = get_the_post_thumbnail_url( $post, 'full' );
+			$attachment_id = get_post_thumbnail_id( $post );
+		}
+
+		if ( ! $podcast_image || ! $this->is_valid_podcast_image( $attachment_id ) ) {
+			$podcast_image = '';
+		}
+
+		return $podcast_image;
+	}
+
+	/**
+	 * @param int $attachment_id
+	 *
+	 * @return bool
 	 */
-	public function get_featured_image( $post ) {
-		return get_the_post_thumbnail_url( $post->ID, 'full' );
+	public function is_valid_podcast_image( $attachment_id ) {
```

Comment:
Nice work. I do think that this could be moved somewhere more "generic" but it's fine here for now.